### DISCUSSION TITLE
[medusa-plugin-auth] Send Bearer token instead of Session cookie authentication

### DISCUSSION THREAD

**Discussion by bacarybruno** [adrien2p/medusa-plugins#98] [Ideas]
Hello @adrien2p, thanks for the amazing work 🙏

Is there a plan to update the `medusa-plugin-auth` plugin to match the changes regarding the authentication that have been done in the latest medusa version ([v1.17.0](https://github.com/medusajs/medusa/releases/tag/v1.17.0))?

I'm thinking specifically about retrieving a Bearer token instead of session cookie to use it for subsequent requests.

Thanks.

**Comment by adrien2p** [comment]
The token is generated following the previous convention of medusa, we need first to align to the new version and then we can on success return it as well wdyt?

**Reply by adrien2p** [reply]
@dPreininger volonteer to do it as he is the one who worked on this update, but as i mentioned it shouldn't be much work I think.

**Reply by dPreininger** [reply]
I think it may be best to both register the session as authenticated and return an access token as well.

I will try to submit a PR tommorow.

**Reply by bacarybruno** [reply]
Sounds like a great plan yes 🎉
